packages/amplify-graphql-api-construct/src/__tests__/__functional__/disable-gen1-patterns.test.ts

```diff
@@ -161,6 +161,22 @@ describe('Deprecate Gen 1 patterns', () => {
     );
   });
 
+  test('does not allow implicit fields on @hasMany', () => {
```

Comment:
This was an opportunistic test addition that I found while I was investigating how a customer-provided GraphQL schema is being mutated in other contexts.